x/proof/keeper/msg_server_submit_proof.go

```diff
@@ -191,3 +202,110 @@ func (k Keeper) deductProofSubmissionFee(ctx context.Context, supplierOperatorAd
 
 	return nil
 }
+
+// proofRequirementForClaim checks if a proof is required for a claim.
+// If it is not, the claim will be settled without a proof.
+// If it is, the claim will only be settled if a valid proof is available.
+// TODO_BLOCKER(@olshansk, #419): Document safety assumptions of the probabilistic proofs mechanism.
+func (k Keeper) ProofRequirementForClaim(ctx context.Context, claim *types.Claim) (_ types.ProofRequirementReason, err error) {
+	logger := k.logger.With("method", "proofRequirementForClaim")
+
+	var requirementReason = types.ProofRequirementReason_NOT_REQUIRED
+
+	// Defer telemetry calls so that they reference the final values the relevant variables.
+	defer func() {
+		telemetry.ProofRequirementCounter(requirementReason, err)
+	}()
+
+	// NB: Assumption that claim is non-nil and has a valid root sum because it
+	// is retrieved from the store and validated, on-chain, at time of creation.
+	var numClaimComputeUnits uint64
+	numClaimComputeUnits, err = claim.GetNumComputeUnits()
+	if err != nil {
+		return requirementReason, err
+	}
+
+	proofParams := k.GetParams(ctx)
+
+	// Require a proof if the claim's compute units meets or exceeds the threshold.
+	//
+	// TODO_BLOCKER(@bryanchriswhite, #419): This is just VERY BASIC placeholder logic to have something
+	// in place while we implement proper probabilistic proofs. If you're reading it,
+	// do not overthink it and look at the documents linked in #419.
+	//
+	// TODO_IMPROVE(@bryanchriswhite, @red-0ne): It might make sense to include
+	// whether there was a proof submission error downstream from here. This would
+	// require a more comprehensive metrics API.
+	if numClaimComputeUnits >= proofParams.GetProofRequirementThreshold() {
+		requirementReason = types.ProofRequirementReason_THRESHOLD
+
+		logger.Info(fmt.Sprintf(
+			"claim requires proof due to compute units (%d) exceeding threshold (%d)",
+			numClaimComputeUnits,
+			proofParams.GetProofRequirementThreshold(),
+		))
+		return requirementReason, nil
+	}
+
+	earliestProofCommitBlockHash, err := k.getEarliestSupplierProofCommitBlockHash(ctx, claim)
```

Comment:
@red-0ne Can you double check that we reject proofs that are too early please>

Reply:
Yes it is, `k.validateProofWindow` is called before `k.ProofRequirementForClaim` in `k.SubmitProof`